sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -98,9 +100,14 @@ case class InMemoryRelation(
     buildBuffers()
   }
 
-  def recache(): Unit = {
-    _cachedColumnBuffers.unpersist()
+  def unpersist(blocking: Boolean = true): Unit = {
+    batchStats.reset()
```

Comment:
To clarify, is this because the content of the batches might change after recomputation in such a way that the use of these batch stats for whole partition pruning would be invalid? It's my understanding that the _set_ of values in each RDD partition will be the same although their order within that partition may change unless a sort is performed (this is the case for reduce tasks due to interleaving of fetched map output, for example).

Given this, it seems like the correctness case that we'd have to worry about is a situation where the old batch stats would have pruned a partition but that pruning decision is invalid with the new stats. But I'm not sure how that can be the case given that pruning decisions seem to be based on conditions defined over the maximum or minimum values of columns and we're effectively constructing partition-wide stats by `AND`-ing conditions over the per-batch stats.

Basically, I think that I see the motivation for this but I don't have an immediate counterexample to show how things would break if we omitted this.